### PR TITLE
Bug 1492322 - Update documentation

### PR DESCRIPTION
**Pull request by staktrace**
Did a pass over the documentation and fleshed out some things. I couldn't figure out what the peekRange stuff in the analysis files was for, does anybody know? As far as I can tell it is not used anywhere.

### CODE REVIEW

**Comment by staktrace** [comment]
I found this commit in the history: `Merge branch 'peek-range' of https://github.com/asutherland/searchfox into peek` so maybe @asutherland can provide more info on what this peekRange stuff is about?

**Comment by asutherland** [comment]
peekRange evolved out of my efforts on https://github.com/mozsearch/mozsearch/pull/11 to include doxygen comments in the context menu so you could easily see docs.

peekRange exists for definitions only and is intended to provide the line number range for both any preceding comment block as well as the entirety of the definition block instead of just the first line.

At the work week I hacked on a solution building on billm's peekRange implementation to show the entirety of the definition match in the search results (without changing the context menu).  My changes from the work week are on the https://github.com/asutherland/searchfox/tree/ui-show-peeklines branch and specifically stashed in the https://github.com/asutherland/searchfox/commit/6b1cc315e9bb71ba25d946c622564a9facb73156 commit.  I got hung up initially by library issues around parsing the query string.  I think it was something like the old hyper library had a more powerful URL class than the new one, and we were moving to the new hyper from the old hyper, but then that got backed out.

Note that in that commit I had changed the implementation of peekRange for efficiency.  Also, I was pursuing some scope creep where I thought it would be nice to syntax highlight the results although billm didn't think we needed to do that.

And in the end I back-burnered it as I wanted to look into the hyper issue and then I somewhat shelved it as I began to pursue various experiments/prototypes on top of searchfox that only recently have begun to come to fruition.

**Comment by staktrace** [comment]
Ok, thanks for the explanation! So I'll leave my updated doc as-is, stating that peekRange is currently unused.

File: docs/analysis.md [hunk 1]
```diff
@@ -113,9 +117,23 @@ index of the field starting from 0.
...
 
+Note that in some cases, the symbol for a C++ identifier might vary from one
+platform to another. For example, a function signature that includes `uint64_t`
+produces a different mangled name on macOS and Linux. In such cases, if
+analysis records are generated for multiple platforms, the final source records will
+contain the symbols from all the platforms merged into a comma-separated list. The
+`scripts/merge-analyses.py` script is used to do this.
+
+Also, in many cases (notably for C++ and JS code), local variables have their
+target records omitted, and their source records have the the `no_crossref`
```
[the remark this answers is not in the document]
Comment (staktrace) [reply docs/analysis.md]: fixed, thanks!

**Pull request merged**